### ISSUE TITLE
[DEV] nodeset report strange error when DNS is not configured well on service node

### ISSUE THREAD

**Issue by robin2008** [xcat2/xcat-core#3787]
When doing nodeset in a hierarchical environment,  found below error:

```
nodeset fake1 osimage=rhels7.4-Alpha1-ppc64le-install-compute -V
fake1: [briggs01]: install rhels7.4-Alpha1-ppc64le-compute
fake1: [sn02]: install rhels7.4-Alpha1-ppc64le-compute
Error: [sn02]: Unable to dispatch hierarchical sub-command to sn01:3001.  Error: Connection failure: IO::Socket::INET: Bad hostname 'sn01:3001' at /opt/xcat/lib/perl/xCAT/Client.pm line 248.
.
fake1: [sn01]: install rhels7.4-Alpha1-ppc64le-compute
Error: [sn01]: Unable to dispatch hierarchical sub-command to sn02:3001.  Error: Connection failure: IO::Socket::INET: Bad hostname 'sn02:3001' at /opt/xcat/lib/perl/xCAT/Client.pm line 248.
```

This could be re-produced when site.sharedtftp=0 and DNS is not configured well on service node (SN cannot resolve name for each other)

**Comment by robin2008** [comment]
Check the logs in /var/log/xcat/cluster.log, and found that the error is reported when dhcp plugin to dispatch the request to other service nodes.

Obviously, it is not work as our design.  when sharedtftp is disabled. nodeset should call `makedhcp -l`, not require makedhcp to dispatch again.

**Comment by robin2008** [comment]
It is fixed by PR #3788
